doc/reference/image_format.md

```diff
@@ -0,0 +1,153 @@
+For example:
+
+```yaml
+templates:
+  /etc/hosts:
+    when:
+      - create
+      - rename
+    template: hosts.tpl
+    properties:
+      foo: bar
+  /etc/hostname:
+    when:
+      - start
+    template: hostname.tpl
+  /etc/network/interfaces:
+    when:
+      - create
+    template: interfaces.tpl
+    create_only: true
+```
+
+The `when` key can be one or more of:
+
+- `create` - run at the time a new instance is created from the image
+- `copy` - run when an instance is created from an existing one
+- `start` - run every time the instance is started
+
+The `template` key points to the template file in the `templates/` folder.
+
+You can pass user-defined template properties to the template file through the `properties` key.
+
+Set the `create_only` key if you want LXD to create the file if it doesn't exist, but not overwrite an existing file.
+
+#### Template files
+
+Template files use the [Pongo2](https://www.schlachter.tech/solutions/pongo2-template-engine/) format.
+
+They always receive the following context:
+
+| Variable     | Type                           | Description                                                                         |
+|--------------|--------------------------------|-------------------------------------------------------------------------------------|
+| `trigger`    | `string`                       | Name of the event that triggered the template                                       |
+| `path`       | `string`                       | Path of the file that uses the template                                             |
+| `instance`   | `map[string]string`            | Key/value map of instance properties (name, architecture, privileged and ephemeral) |
+| `config`     | `map[string]string`            | Key/value map of the instance's configuration                                       |
+| `devices`    | `map[string]map[string]string` | Key/value map of the devices assigned to the instance                               |
+| `properties` | `map[string]string`            | Key/value map of the template properties specified in `metadata.yaml`               |
+
+For convenience, the following functions are exported to the Pongo2 templates:
+
+- `config_get("user.foo", "bar")` - Returns the value of `user.foo`, or `"bar"` if not set.
+
+## Image tarballs
+
+LXD supports two LXD-specific image formats: a unified tarball and split tarballs.
+
+These tarballs can be compressed.
+LXD supports a wide variety of compression algorithms for tarballs.
+However, for compatibility purposes, you should use `gzip` or `xz`.
+
+(image-format-unified)=
+### Unified tarball
+
+A unified tarball is a single tarball (usually `*.tar.xz`) that contains the full content of the image, including the metadata, the root file system and optionally the template files.
+
+This is the format that LXD itself produces.
+Therefore, you should use a unified tarball for LXD-specific images.
```

Comment:
This comes from the existing documentation. But when I use `lxc image export`, I actually get a split image.
Is the statement wrong? Or does it refer to the images LXD provides on the images server? (But those look like split images to me as well?)

Reply:
I think this is referring to the format of the image produced when using `lxc publish`, as doing that and then using `lxc image export` downloads a single tarball file.

So `lxc image export` seems to only be exporting the image in its current format (whatever that was from the remote server or locally published) and not "producing" it. It is unclear though I agree.

Reply:
OK, reworded a bit - and I removed the recommendation since we're apparently not following it ourselves. ;)

Reply:
@stgraber what do you think about this?

Reply:
`This is the format that LXD itself produces.` is indeed still accurate.
LXD does not produce the images it downloads from the various image servers, those are produced by tools like distrobuilder. LXD produces images when one uses `lxc publish` and when one does that, the result is indeed a unified image.

The recommendation still mostly stands that if you're going to be building images by hand for consumption by LXD alone, then you should go with the unified format as it's far easier to understand and far harder to mess up.

Using the split format is very convenient for cases where images may be consumed by other tools than LXD.
That's why the official Ubuntu images are in that format (consumed by Docker, LXD, ...) as well as why our own images are also in that format (consumed by both LXC and LXD).

Reply:
Updated as follows for the unified tarballs:
"This is the format that LXD itself uses internally when publishing images. It is usually easier to work with; therefore, you should use the unified format when creating LXD-specific images."

And for the split tarballs:
"This format is designed to allow for easy image building from existing non-LXD rootfs tarballs that are already available. You should also use this format if you want to create images that can be consumed by both LXD and other tools."

Reply:
Sounds good!